modules/javafx.controls/src/main/java/javafx/scene/control/skin/TextAreaSkin.java

```diff
@@ -175,10 +175,12 @@ public TextAreaSkin(final TextArea control) {
             }
         };
         caretPosition.addListener((observable, oldValue, newValue) -> {
+            setCaretAnimating(false);
             targetCaretX = -1;
             if (control.getWidth() > 0) {
                 setForwardBias(true);
             }
+            setCaretAnimating(true);
```

Comment:
I find this a bit confusing.  By putting the two `setCaretAnimating` separate, you give the impression it must stop blinking for a while before continuing because of the call to `setForwardBias`.

However, all we want to is call it with `false` then `true` to restart the animation (for a lack of a `playFromStart` call).  Perhaps it may be good to introduce a method that captures this more clearly as I find it confusing (`restartCaretAnimation`)

Same comment for `TextFieldSkin`.

Reply:
good point